homeassistant/components/thermostat/honeywell.py

```diff
@@ -23,20 +25,25 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
-
+    try:
+        away_temp = float(config.get(CONF_AWAY_TEMP,16))
+    except ValueError:
+        _LOGGER.error("value entered for item {} should convert to a number"
+                      .format(CONF_AWAY_TEMP))
```

Comment:
Make sure you return `False` after finding an error to prevent further initialization .